src/extensibility/ExtensionManager.js

```diff
@@ -398,6 +408,60 @@ define(function (require, exports, module) {
         }
         return result.promise();
     }
+    
+    /**
+     * Disables the installed extension with the given id.
+     * 
+     * @param {string} id The id of the extension to disable.
+     * @return {$.Promise} A promise that's resolved when the extenion is disabled or
+     *      rejected with an error that prevented the disabling.
+     */
+    function disable(id) {
+        var result = new $.Deferred(),
+            extension = extensions[id];
+        if (extension && extension.installInfo) {
+            Package.disable(extension.installInfo.path)
+                .done(function () {
+                    extension.installInfo.status = DISABLED;
+                    extension.installInfo.metadata.disabled = true;
+                    result.resolve();
+                    exports.trigger("statusChange", id);
+                })
+                .fail(function (err) {
+                    result.reject(err);
+                });
+        } else {
+            result.reject(StringUtils.format(Strings.EXTENSION_NOT_INSTALLED, id));
+        }
+        return result.promise();
+    }
+    
+    /**
+     * Enables the installed extension with the given id.
+     * 
+     * @param {string} id The id of the extension to enable.
+     * @return {$.Promise} A promise that's resolved when the extenion is enabled or
+     *      rejected with an error that prevented the enabling.
+     */
+    function enable(id) {
+        var result = new $.Deferred(),
```

Comment:
If we had a single function in `Package`, I would undoubtedly do something like what you guys suggest. However, I am reluctant to sacrifice "compactness" of the code for readability. Firstly, the code will have to call `Package.(enable|disabled)` via variable; secondly, instead of using clear literals (`true`, `false`, `DISABLED`, `ENABLED`) variables should be used. I don't see how it makes the code more readable.